src/lib/mina_base/test/account/account.ml

```diff
@@ -0,0 +1,273 @@
+       are completely vested before the largest possible slot, (2^32)-1. *)
+    let%test_unit "Test fine-tuning of the account generation." =
+      Quickcheck.test Account.gen_timed ~f:(fun account ->
+          [%test_eq: Balance.t option] (Some Balance.zero)
+            ( match account.timing with
+            | Untimed ->
+                None
+            | Timed
+                { initial_minimum_balance
+                ; cliff_time
+                ; cliff_amount
+                ; vesting_period
+                ; vesting_increment
+                } ->
+                let global_slot = Global_slot.max_value in
+                Option.some
+                @@ Account.min_balance_at_slot ~global_slot ~cliff_time
+                     ~cliff_amount ~vesting_period ~vesting_increment
+                     ~initial_minimum_balance ) )
+
+    let%test_unit "Minimum balance never changes before the cliff time." =
+      Quickcheck.test
+        (let open Quickcheck.Generator.Let_syntax in
+        let%bind timing = gen_timing in
+        let max_global_slot =
+          Global_slot.(sub timing.cliff_time (of_int 1)) |> Option.value_exn
+        in
+        let%map global_slot = Global_slot.(gen_incl zero max_global_slot) in
+        (timing, global_slot))
+        ~f:(fun (timing, global_slot) ->
+          [%test_eq: Balance.t] timing.initial_minimum_balance
+            (min_balance_at_slot timing ~global_slot) )
+
+    let%test_unit "Cliff amount is immediately released at cliff_time." =
+      Quickcheck.test gen_timing ~f:(fun timing ->
+          let min_balance =
+            Balance.(timing.initial_minimum_balance - timing.cliff_amount)
+            |> Option.value_exn
+          in
+          [%test_eq: Balance.t] min_balance
+            (min_balance_at_slot timing ~global_slot:timing.cliff_time) )
+
+    let%test_unit "Minimum balance never increases over time." =
+      Quickcheck.test
+        (let open Quickcheck.Generator.Let_syntax in
+        let%bind timing = gen_timing in
+        let%bind slot1 = Global_slot.gen in
+        let%map slot2 = Global_slot.(gen_incl slot1 max_value) in
+        (timing, slot1, slot2))
+        ~f:(fun (timing, slot1, slot2) ->
+          [%test_pred: Balance.t * Balance.t]
+            (Tuple.T2.uncurry Balance.( >= ))
+            ( min_balance_at_slot timing ~global_slot:slot1
+            , min_balance_at_slot timing ~global_slot:slot2 ) )
+
+    let%test_unit "Final vesting slot is the first when minimum balance is 0." =
+      Quickcheck.test gen_timing ~f:(fun timing_rec ->
+          let timing = Account.Timing.of_record timing_rec in
+          let vesting_final_slot = Account.timing_final_vesting_slot timing in
+          [%test_eq: Balance.t] Balance.zero
+            (min_balance_at_slot timing_rec ~global_slot:vesting_final_slot) ;
+          let one_slot_prior =
+            Global_slot.(sub vesting_final_slot (of_int 1))
+            |> Option.value ~default:Global_slot.zero
```

Comment:
This will make the generator fail if the last vesting slot is 0. That shouldn't happen with the quickcheck generator, but because of this, I think it's better to just `Option.value_exn` here. That way, if someone breaks the generator in the future, the error they get is more actionable.

Reply:
Done in 94748d2b75.